client-side/src/components/signUp/signUp.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import { useDispatch } from 'react-redux';
+import ReCAPTCHA from 'react-google-recaptcha';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+
+import './signUp.scss';
+import PasswordStrengthMeter from '../signUp/PasswordStrength';
+import GenericButton from '../../stories/Button/GenericButton';
+import GenericInput from '../../stories/GenericInput/genericInput';
+import MESSAGES from './constants';
+import { useNavigate } from 'react-router-dom';
+import { createUser } from '../../services/userService';
+import { addUser } from '../../redux/user/user.slice';
+
+
+
+const SignUpSchema = Yup.object().shape({
+  name: Yup.string()
+    .matches(/^[a-zA-Z\u0590-\u05FF\s]+$/, MESSAGES.name.matches)
+    .required(MESSAGES.name.required),
+  email: Yup.string()
+    .email(MESSAGES.email.invalid)
+    .required(MESSAGES.email.required),
+  password: Yup.string()
+    .required(MESSAGES.password.required)
+    .matches(/[A-Za-z]/, MESSAGES.password.matches.letters)
+    .matches(/\d/, MESSAGES.password.matches.digits)
+    .min(4, MESSAGES.password.min)
+});
+function SignUp() {
+  const [password, setPassword] = useState('');
+  const dispatch = useDispatch();
+  const formik = useFormik({
+    initialValues: { name: '', email: '', password: '' },
+    validationSchema: SignUpSchema,
+    onSubmit: (values) => {
+      userSignUp({
+        name: values.name,
```

Comment:
Why are you assigning the values object to a new object? they are exactly the same